### PR TITLE
🖍🚀🐛support for CAaja ads network

### PR DESCRIPTION
**Pull request by matsuda-takehiro**
image
```Can not ad transition because of "absolute" of <div id = "C">.```

Suggested revision above

### CODE REVIEW

**Comment by watralyaf** [comment]
شسلت

**Comment by watralyaf** [comment]
شكرا

**Comment by matsuda-takehiro** [comment]
@zhouyx @torch2424 
Thank you for the last time.
This time I wanted to make a few adjustments, and I did PR.
Please review please

**Comment by matsuda-takehiro** [comment]
Is it better to use
｀｀｀global.document.getElementById('c').appendChild(elStyle);｀｀｀
instead of
｀｀｀global.document.body.appendChild(elStyle);｀｀｀
?

**Comment by matsuda-takehiro** [comment]
How do I run draw3p ()?

hmm..
https://github.com/ampproject/amphtml/issues/6694
image

**Comment by matsuda-takehiro** [comment]
@zhouyx @torch2424
Please give me a chance again😭

**Comment by torch2424** [comment]
@matsuda-takehiro 

Hello! Sorry for the late(r) review. Was busy with the upcoming AMP Conf event! And then the weekend passed by.

> Is it better to use `global.document.getElementById('c').appendChild(elStyle);` instead of `global.document.body.appendChild(elStyle);`?

I personally don't have enough context on what you are trying to do. Could you explain that? It's up to you, as it will only affect your customers who are using your `amp-ad` type.

That being said, all of the changes look goo to me 😄 Let me know if you want to make anymore, or you want me to just merge this in 👍

**Comment by matsuda-takehiro** [comment]
@torch2424 
Thank you!!
It was an event in Tokyo, right?
I can also listen to Japanese by simultaneous translation and listen in real time on the Internet, and I look forward to it!

I am sorry while busy.

I will summarize it below!

**Comment by matsuda-takehiro** [comment]
image

Now, aja.js is creating an iframe, but an element of id = c is attached to cover the iframe, so I thought to make it a child element of id = c.

image

**Comment by matsuda-takehiro** [comment]
https://amp.dev/
It has become a very easy-to-see and cool site!👍👍

**Comment by matsuda-takehiro** [comment]
If you can see and click,
Sorry to trouble you, but please merge it!

File: ads/aja.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ export function aja(global, data) {
     sspCode: data['sspCode'],
   });
 
+  const elAttri = 'allow-same-origin allow-top-navigation ';
+  const elAttri2 = 'allow-scripts allow-popups ';
```
Comment (calebcordry) [inline ads/aja.js]: Are you sure you want to add all of these? They will only apply to the inner iframe, not the top level iframe. Is that OK?

Comment (matsuda-takehiro) [reply ads/aja.js]: Thank you very much. Maybe it's okay, is there a problem?

**Comment by matsuda-takehiro** [comment]
This change has no impact on users.
Because I haven't used it in the first place😭

**Comment by matsuda-takehiro** [comment]
I modified sandbox attributes and I think it maybe works well.

Also I changed interface of our request parameter.
But it affects to no users, because currently no customers is using.

**Comment by matsuda-takehiro** [comment]
travis failed. What should I do?

**Comment by 0tarof** [comment]
Hi there.
I am working with @matsuda-takehiro .
I found issue about rendering our ads. So we have to fix rendering code.
We'll push another commits tomorrow in JST.
So then, please wait until after the commits.

Thanks.

**Comment by torch2424** [comment]
@matsuda-takehiro 

Thanks for making @calebcordry changes 😄 

So it seems that "Taro FURUKAWA" has also committed to this branch, and hasn't sign the CLA. Could you reach out to them, and have the email which made the commit to get the CLA signed? Or if this was a mistake, you may be able to rebase your branch. The first option is probably easier 😄 

Also, for the travis, please see the logs: https://travis-ci.org/ampproject/amphtml/jobs/518288450

Seems like you used `getComputerStyle`, and you should import our cross browser friendly solution from: `style.js`. See the logs as it provides more details 👍 

Let me know if this helps, thanks!

**Comment by matsuda-takehiro** [comment]
We found ad rendering issue.
And my co-worker has resolved this.

**Comment by 0tarof** [comment]
@torch2424 Thank you for reviewing.
I replaced `getComputedStyle` with `computedStyle` from `style.js`.

Thanks.

**Comment by matsuda-takehiro** [comment]
@torch2424 
thank you!!

We found ad rendering issue.
And my co-worker has resolved this.

**Comment by matsuda-takehiro** [comment]
test

**Comment by 0tarof** [comment]
@torch2424 
I've fixed all issues depends on my code.
But still failed on remote test.
Is there anything that I should do?

**Comment by torch2424** [comment]
@matsuda-takehiro , "Taro FURUKAWA"

Thanks for making the changes. The last step would to be to have "Taro FURUKAWA" sign the CLA 😄 Please see the comment from @googlebot . You would need to sign with the same email you used to make the commits, usually set up with `git config`.

Let me know if this helps. Thanks! 😄

**Comment by 0tarof** [comment]
@torch2424 
Thank you for reviewing.

> The last step would to be to have "Taro FURUKAWA" sign the CLA 
I signed after first @googlebot comment, and @googlebot confirm my sign of CLA at https://github.com/ampproject/amphtml/pull/21671#issuecomment-481925115 .

And also said "It's up to you to confirm consent of all the commit author(s), set the cla label to yes (if enabled on your project)".
So I think next step is you or other maintainer set cla label to yes, isn't it?

**Comment by matsuda-takehiro** [comment]
@torch 2424
I am always grateful for your help.
Will I have something to do?

**Comment by torch2424** [comment]
@matsuda-takehiro Hello!

Sorry for the delay, currently helping out at AMP Conf Tokyo. Will take a look once we all get back next week.

Thank you! 😄

**Comment by matsuda-takehiro** [comment]
@torch 2424
Thank you for the very interesting event!
amp-script looks interesting.
Sorry for your inconvenience but I would like a review

Thank you!

**Comment by torch2424** [comment]
@matsuda-takehiro 

Some of us are back from AMP Conf. Seems like the CLA is in order now 😄 

Going to cc @calebcordry and/or @lannka for the final review. As it seems like there were a lot of changes made between commits form the original one. 👍 

Thanks again!

**Comment by lannka** [comment]
@matsuda-takehiro thanks for the PR. 

The code looks good to me in general. I saw you added a bunch of logic handling no-fill and re-size. If possible, I'd suggest to move more logic into your own js file, instead of having them checked in AMP repo. In that way it's more flexible for your future updates and also keeping AMP library slim.

**Comment by 0tarof** [comment]
@lannka 
Thank you for review and advice.
I remove logic to our own js and keep only loading script.

Please check it again.

**Comment by 0tarof** [comment]
@lannka 
> please also make changes to /integration.js and amp-ad.md, and test it locally.

We've already changed these files in previous PR(https://github.com/ampproject/amphtml/pull/16306).

But we currently use `<amp-ad>` instead of `<amp-embed>`, so I moved our link from embed types to ad networks.

*IMPORTANT*
This PR includes breaking changes from previous version.
But it affects to no users, because currently no customers are using our network on AMP page.

Thanks,

**Comment by matsuda-takehiro** [comment]
@lannka @torch2424 
Thank you !
Please merge our PR.
Is there anything I need to do?

**Comment by torch2424** [comment]
@matsuda-takehiro I re-ran the travis to get the flake to pass, and I'll let @lannka do the merge, as they approved 😄

**Comment by matsuda-takehiro** [comment]
@torch2424 @lannka 
Thank you !

If it is good, please let me know the implementation time! Last time was about two weeks later

**Comment by torch2424** [comment]
@matsuda-takehiro Yep! You are correct about the Release Schedule.

Around Thursday, you should be able to test this in the Dev Channel.

And around the Thursday after that, it should be in PROD.

Thank you! 😄

**Comment by matsuda-takehiro** [comment]
@torch2424
Thank you !
I'm looking forward to it!🥰

**Pull request merged**